### PR TITLE
[Nextjs]Upgrade 3rd party dependencies

### PR DESCRIPTION
**Pull request by illiakovalenko**
## Description / Motivation
* Upgraded 3rdy party libraries to latest versions except:
  * `graphql` and `@graphql-codegen/cli@^2.0.0` - has `graphql-tools` -> `@whatwg-node/fetch` -> `undici` and imported files not resolved in the right way by webpack
  image
  * `chalk@^5.0.0` can't be installed since it doesn't include cjs build anymore, but ^4.0.0 is fine to use (all features are the same)
  * `sass-alias@^2.0.0` not compatible with nextjs sass webpack configuration
* Now _graphql-let_ generates types under `node_modules/@types/graphql-let` folder
* Prettier typescript eslint rules are bundled into `prettier` extension
* Enzyme _react 18_ warnings are suppressed since there is no new version for upgrade
* Removed extra _axios_ dependency and exposed _AxiosResponse_ from _sitecore-jss_ instead
* Reimplemented tests for _redirects_middleware_, they didn't validate response and had been accepting invalid _redirectType_
* Updated class names according to breaking changes provided by _bootstrap 5_

## Testing Details
* Manual testing
* Ran `yarn install`, `npm install`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by art-alexeyenko — APPROVED** [review]
Looking good!

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/package.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
   "dependencies": {
-    "bootstrap": "^5.1.3",
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/templates/nextjs-sxa/package.json:3]: We probably want to leave `nextjs-sxa` alone and leave it up to the SXA team to update/test these dependencies?

**Review by ambrauer — APPROVED** [review]
Looks good to me, just one comment

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/package.json (hunk 1 above)
Comment (illiakovalenko) [reply packages/create-sitecore-jss/src/templates/nextjs-sxa/package.json]: Makes sense for me, I was thinking about that

**Pull request merged**